aws-codeguruprofiler-profilinggroup/src/test/java/software/amazon/codeguruprofiler/profilinggroup/UpdateHandlerTest.java

```diff
@@ -58,10 +69,206 @@
+                            software.amazon.codeguruprofiler.profilinggroup.Channel.builder()
+                                    .channelUri("channelUri2")
+                                    .build()))
+                    .build());
+
+            subject.handleRequest(proxy, request, null, logger);
+
+            verify(proxy, times(1)).injectCredentialsAndInvokeV2(eq(getPolicyRequest), any());
+            verify(proxy, times(1)).injectCredentialsAndInvokeV2(eq(getNotificationConfigurationRequest), any());
+            verify(proxy, times(1)).injectCredentialsAndInvokeV2(eq(removeNotificationChannelRequest), any());
+            verify(proxy, times(1)).injectCredentialsAndInvokeV2(eq(AddNotificationChannelsRequest.builder().profilingGroupName(profilingGroupName).channels(
+                    Channel.builder().id("channelIdNew").uri("channelUri").eventPublishers(EventPublisher.ANOMALY_DETECTION).build())
+                            .build()), any());
+            verifyNoMoreInteractions(proxy);
+
+        }
+
+        @Test
+        public void itDoesNotAddOrDeleteNotificationChannelWithNoChangeSet() {
+            doReturn(GetNotificationConfigurationResponse.builder()
+                    .notificationConfiguration(NotificationConfiguration.builder()
+                            .channels(Channel.builder()
+                                            .id("channelId")
+                                            .uri("channelUri")
+                                            .eventPublishers(EventPublisher.ANOMALY_DETECTION)
+                                            .build(),
+                                    Channel.builder()
+                                            .id("channelId2")
+                                            .uri("channelUri2")
+                                            .build())
+                            .build())
+                    .build())
+                    .when(proxy).injectCredentialsAndInvokeV2(eq(getNotificationConfigurationRequest), any());
+
+            request = makeRequest(ResourceModel.builder().profilingGroupName(profilingGroupName).anomalyDetectionNotificationConfiguration(
+                    ImmutableList.of(
+                            software.amazon.codeguruprofiler.profilinggroup.Channel.builder()
+                                    .channelUri("channelUri")
+                                    .build(),
+                            software.amazon.codeguruprofiler.profilinggroup.Channel.builder()
+                                    .channelUri("channelUri2")
+                                    .build()))
+                    .build());
+
+            subject.handleRequest(proxy, request, null, logger);
+
+            verify(proxy, times(1)).injectCredentialsAndInvokeV2(eq(getPolicyRequest), any());
+            verify(proxy, times(1)).injectCredentialsAndInvokeV2(eq(getNotificationConfigurationRequest), any());
+            verifyNoMoreInteractions(proxy);
+        }
```

Comment:
This seems like a duplicate of the first test, `testSuccess`. It does not seem to differ from it, other than using two channels instead of 1?

Reply:
Must have been a bad rebase, the `testSuccess` method was renamed to this in a previous commit.